plugins/Marketplace/vue/src/Marketplace/Marketplace.vue

```diff
@@ -23,251 +23,233 @@
-      <form
-        method="post"
-        class="plugin-search"
-        :action="pluginSearchFormAction"
-        ref="pluginSearchForm"
-      >
+    <div class="col s12 m12 l4 " v-if="pluginsToShow?.length > 20 || searchQuery">
+      <div class="plugin-search">
         <div>
           <Field
             uicontrol="text"
             name="query"
             :title="queryInputTitle"
             :full-width="true"
-            v-model="searchQuery"
+            :model-value="searchQuery"
+            @update:model-value="updateQuery"
           >
           </Field>
         </div>
         <span
           class="icon-search"
-          @click="$refs.pluginSearchForm.submit()"
         />
-      </form>
+      </div>
     </div>
   </div>
+
+  <PluginList v-if="!loading && pluginsToShow.length > 0"
+              :plugins-to-show="pluginsToShow"
+              :is-auto-update-possible="isAutoUpdatePossible"
+              :is-super-user="isSuperUser"
+              :is-multi-server-environment="isMultiServerEnvironment"
+              :is-plugins-admin-enabled="isPluginsAdminEnabled"
+              :is-valid-consumer="isValidConsumer"
+              :deactivate-nonce="deactivateNonce"
+              :activate-nonce="activateNonce"
+              :install-nonce="installNonce"
+              :update-nonce="updateNonce"
+              :download-nonce="downloadNonce"
+              @triggerUpdate="this.fetchPlugins()"
+  />
+
+  <ContentBlock v-if="!loading && pluginsToShow.length == 0">
+    {{ translate(showThemes ? 'Marketplace_NoThemesFound' : 'Marketplace_NoPluginsFound') }}
+  </ContentBlock>
+
+  <ContentBlock v-if="loading">
+    <img
+      src="plugins/Morpheus/images/loading-blue.gif"
+      :alt="translate('General_LoadingData')"
+    />
+    {{ loadingMessage }}
+  </ContentBlock>
 </template>
 
 <script lang="ts">
-import { defineComponent, nextTick } from 'vue';
+import { defineComponent, watch } from 'vue';
 import {
-  debounce, translate, MatomoUrl, Matomo,
+  translate, Matomo, MatomoUrl, AjaxHelper, ContentBlock,
 } from 'CoreHome';
 import { Field } from 'CorePluginsAdmin';
+import PluginList from '../PluginList/PluginList.vue';
 
 interface MarketplaceState {
+  loading: boolean;
+  fetchRequest: Promise<void>|null;
+  fetchRequestAbortController: AbortController|null;
   pluginSort: string;
   pluginTypeFilter: string;
   searchQuery: string;
+  pluginsToShow: Array<Record<string, unknown>>;
 }
 
 const lcfirst = (s: string) => `${s[0].toLowerCase()}${s.substring(1)}`;
 
-const { $ } = window;
-
 export default defineComponent({
   props: {
-    pluginType: {
-      type: String,
-      required: true,
-    },
     pluginTypeOptions: {
       type: [Object, Array],
       required: true,
     },
-    sort: {
+    defaultSort: {
       type: String,
       required: true,
     },
     pluginSortOptions: {
       type: [Object, Array],
       required: true,
     },
-    pluginsToShow: {
+    numAvailablePlugins: {
```

Comment:
We should come up with a better name, it's not clear it holds an array of numbers for each purchase type.

Reply:
Renamed it to `numAvailablePluginsByType`. Or do you have a suggestion for a more suitable name/structure?

Reply:
This new name is good, thanks!